src/app/application/components/app-collection/AppCollection.vue

```diff
@@ -89,6 +91,15 @@ const slots = useSlots()
 
 const items = ref<unknown[] | undefined>(props.items)
 const cacheKey = ref<number>(0)
+/**
+ * Used as a means to instruct KTable to re-mount.
+ *
+ * This is a hack around the fact that there is no other way to tell KTable if the current page of a paginated view has changed. KTable assumes it’s the *sole* owner/maintainer of pagination-related state, but that’s a design flaw that doesn’t account for browser history navigation (e.g. pressing back to go from page 2 to page 1).
+ *
+ * Is triggered via a watcher whenever `props.pageNumber` changes while the last page number that was emitted by KTable (via calls to its `fetcher` prop) is different (i.e. the page number has changed independently of a KTable mechanism).
+ */
+const kTableMountKey = ref(0)
+const lastPageNumber = ref(props.pageNumber)
```

Comment:
**Comment**: Personally, I think the better solution to this would be supporting a new `props.pageNumber` in KTable. Adding a watcher to it and updating the internal `page` variable whenever it is set _to a different value than `page`_ would trigger the same mechanisms attached to using the built-in pagination and consequently trigger `props.fetcher` with the correct parameters.

KTable is currently the sole owner of all matters of pagination and I think that’s a design flaw that should be considered a bug rather than a lack of a feature as browser history navigation is a core browser feature that it currently can’t handle (except for via the hack employed here).

Reply:
so why don't we just fix it in KTable?

Reply:
@kleinfreund to link a PR here until it's merged we will use this workaround

Reply:
I’ve updated the in-code comment with the link in https://github.com/kumahq/kuma-gui/commit/35eafe0a6a1678ecf3d71d99f0a9ff61821076cc